Sources/App/routes.swift

```diff
@@ -105,22 +105,40 @@ func routes(_ app: Application) throws {
         
         
         @Sendable
-        func initializeSession(webSocket: WebSocket) {
+        func initializeSession(webSocket: WebSocket) async {
             do {
-                let sessionConfigJSONString = Constants.loadSessionConfig()
+                await serviceState.initializeCurrentService()
+                let initialQuestion = await serviceState.current.getNextQuestion()
+                let initialSystemMessage = await Constants.getSystemMessageForService(
+                    serviceState.current,
+                    initialQuestion: initialQuestion ?? "No question found."
+                )
+                let sessionConfigJSONString = Constants.loadSessionConfig(systemMessage: initialSystemMessage ?? "")
                 req.logger.info("\(sessionConfigJSONString)")
-                webSocket.send(sessionConfigJSONString)
+                try await webSocket.send(sessionConfigJSONString)
                 let responseRequest: [String: Any] = [
                     "type": "response.create"
                 ]
                 let responseData = try JSONSerialization.data(withJSONObject: responseRequest)
                 if let jsonString = String(data: responseData, encoding: .utf8) {
-                    webSocket.send(jsonString)
+                    try await webSocket.send(jsonString)
                 }
             } catch {
                 req.logger.error("Failed to serialize session update: \(error)")
             }
         }
+        
+        @Sendable
+        func updateSession(webSocket: WebSocket, systemMessage: String) async {
+            let sessionConfigJSONString = Constants.loadSessionConfig(systemMessage: systemMessage)
+            do {
+                print("Updading session with:")
+                print(sessionConfigJSONString)
+                try await webSocket.send(sessionConfigJSONString)
+            } catch {
+                print("Failed to serialize update request: \(error)")
```

Comment:
We should aim to move all print statements to logger statements + also ensure that we handle the error messages here as expected, would it make sense to close the socket in case we failed to send the request?